test/nettle.jl

```diff
@@ -0,0 +1,32 @@
+import Nettle
+
+function test_equal_state(m::MD5.MD5_CTX,n::Nettle.Hasher)
+    @test (n.state)[1:16] == reinterpret(UInt8, m.state)
+end
+
+@testset "Against Nettle initialize, update!, digest!" begin
+    for chunkcount ∈ [0,1,2,3, 10, 100]
+        m = MD5.MD5_CTX()
+        n = Nettle.Hasher("md5")
+        test_equal_state(m,n)
+        for _ in 1:chunkcount
+            chunksize = rand(0:10000)
+            data = rand(UInt8, chunksize)
+            test_equal_state(m, n)
+            MD5.update!(m, data)
+            Nettle.update!(n, data)
+            test_equal_state(m,n)
```

Comment:
`test_equal_state` runs twice per iteration.

Reply:
true the former can be deleted.

Reply:
true the former can be deleted. I should have caught that.